lib/httpDigest.js

```diff
@@ -1,55 +1,100 @@
 /*!
  * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
  */
-import base64url from './base64url.js';
-import base64 from './base64.js';
+import {encode as base64Encode} from 'universal-base64';
+import {encode as base64urlEncode} from 'base64url-universal';
 import crypto from './crypto.js';
 
 /**
- * Creates a string suitable for HTTP Digest: header.
+ * Creates a value suitable for the HTTP `Digest` header.
  *
- * @param {string|object} data - Objects to be hashed (typically, headers).
- * @param {string} algorithm - Hash algorithm to use. (e.g. 'sha256').
- * @param {boolean} useMultihash - Whether to encode via multihash.
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - Objects to be hashed
+ *   (typically headers).
+ * @param {string} [options.algorithm] - Hash algorithm to use.
+ *   (e.g. 'sha256').
+ * @param {boolean} options.useMultihash - Whether to encode via multihash.
  *   If false, the hash will be base64 encoded (non-url).
  *
- * @returns {Promise<string>} Resolves with hashed digest string.
+ * @returns {Promise<string>} Resolves to `Digest` header value.
  */
-async function create({data, algorithm = 'sha256', useMultihash = true}) {
+export async function createHeaderValue(
+  {data, algorithm = 'sha256', useMultihash = true} = {}) {
+  const {key, encodedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash});
+  return `${key}=${encodedDigest}`;
+}
+
+/**
+ * Verifies the HTTP `Digest` header value against the given HTTP body `data`.
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.headerValue - The digest header value to verify
+ *   the data against.
+ *
+ * @returns {object} - An object with `verified` set to `true` or `false`.
+ */
+export async function verifyHeaderValue({data, headerValue} = {}) {
+  const {key, algorithm, encodedDigest} = _parseHeaderValue(headerValue);
+  const {encodedDigest: expectedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash: key === 'mh'});
+  return {verified: encodedDigest === expectedDigest};
+}
+
+async function _createHeaderValueComponents(
+  {data, algorithm = 'sha256', useMultihash = true} = {}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
-  const encodedData = new TextEncoder().encode(data);
-  let digest;
+  if(algorithm !== 'sha256') {
+    throw new Error(`Algorithm "${algorithm}" is not supported.`);
+  }
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return {key: 'mh', encodedDigest: _createMultihash({digest})};
+  }
+  return {key: 'SHA-256', encodedDigest: base64Encode(digest)};
```

Comment:
I feel like the `key` value being returned here should be informed by the return value of `_getDigest` on L53.  I would like to see an easier transition when the time comes to add additional methods.  Possibly even including string literal `sha256` being in some CONST array of accepted digest methods.

looking further down, i see a possibility that the data structure for a hash algorithm might also include the expected multibase prefix instead of using string literals in code below in `_parseHeaderValue`.